src/Components/ModalList/Auth/EmailConfirm/EmailConfirm.tsx

```diff
@@ -1,16 +1,17 @@
-import React, { createRef, useCallback, useEffect } from "react";
-import { AuthTemplate } from "Components";
-import * as S from "./Style";
-
+import React, { useCallback } from "react";
 import { useModalContext } from "Utils/Contexts/ModalContext";
-import { InfoModal } from "Components";
 import { Envelope } from "Assets";
+import AuthTemplate from "../AuthTemplate/AuthTemplate";
+import InfoModal from "../../InfoModal/InfoModal";
+
+import * as S from "./Style";
 
 const EmailConfirm: React.FC = () => {
   const { addModal, removeModal } = useModalContext();
 
   const numberMaxLength = (e: React.FormEvent<HTMLInputElement>) => {
-    let { maxLength, value } = e.nativeEvent.target as HTMLInputElement;
+    const { maxLength } = e.nativeEvent.target as HTMLInputElement;
```

Comment:
please use the type guard, Avoid duct typing if not necessary

Reply:
Prevents type inference from being made correctly.

Reply:
because I recommend the following code

```tsx
const numberMaxLength = (e: React.ChangeEvent<HTMLInputElement>) => {
	  const { maxLength } = e.target;
	  let { value } = e.target;
	  if (value.length > maxLength) value = value.slice(0, maxLength);
};

<S.PinInput
	  maxLength={1}
	  onInput={numberMaxLength}
	  name="input-1"
	  autoFocus
	  autoCorrect="off"
	  autoComplete="off"
	  autoCapitalize="off"
	  spellCheck="false"
	  type="number"
	  onKeyDown={pressEvent}
	  onChange={autoFocusing} 
/>
```